### PR TITLE
Adjust metric APIs and add tests

### PR DESCRIPTION
**Pull request by pakrym**
Show some love to metrics

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.Query/tests/MetricsSenderClient.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            _clientDiagnostics = new ClientDiagnostics(options);
+        }
+
+        public async Task<Response> SendAsync(object data)
+        {
+            byte[] dataBytes = JsonSerializer.SerializeToUtf8Bytes(data);
+            string s = JsonSerializer.Serialize(data);
+
+            var request = _pipeline.CreateRequest();
+            request.Uri.Reset(new Uri($"https://{_location}.{_ingestEndpointPrefix}{_resourceName}/metrics?api-version=2018-09-01-preview"));
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/tests/MetricsSenderClient.cs:37]: Should the API version be hard coded?

**Review by christothes — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.Query/tests/MetricsSenderClient.cs (hunk 1 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/tests/MetricsSenderClient.cs:37]: I think it's fine for a test utility type.

File: sdk/monitor/Azure.Monitor.Query/tests/MetricsQueryClientLiveTests.cs [hunk 2]
```diff
@@ -33,21 +43,212 @@ public async Task CanListMetrics()
...
+                new[]{ _testData.MetricName },
+                new MetricQueryOptions()
+                {
+                    MetricNamespace = _testData.MetricNamespace,
+                    StartTime = _testData.StartTime,
+                    Duration = TimeSpan.FromMinutes(3)
+                });
+
+            var timeSeriesData = results.Value.Metrics[0].Timeseries[0].Data;
+            Assert.AreEqual(3, timeSeriesData.Count);
```
[the remark this answers is not in the document]
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/tests/MetricsQueryClientLiveTests.cs:62]: It's related to the `TimeSpan.FromMinutes(3)` part. 1 entry per minute with 3 minute duration == 3 entries.

[the remark this answers is not in the document]
Comment (christothes) [reply sdk/monitor/Azure.Monitor.Query/tests/MetricsQueryClientLiveTests.cs:62]: That would be great to express semantically in the assert! 😄

File: sdk/monitor/Azure.Monitor.Query/src/MetricQueryOptions.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        /// Gets or sets the maximum number of records to retrieve. Valid only when <see cref="Filter"/> is specified. Defaults to <c>null</c>.
+        /// </summary>
+        public int? Top { get; set; }
+
+        /// <summary>
+        /// Gets ot sets the  filter is used to reduce the set of metric data returned.
+        /// Example:
+        ///     Metric contains metadata A, B and C.
+        ///
+        ///     - Return all time series of C where A = a1 and B = b1 or b2:
```
[the remark this answers is not in the document]
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/MetricQueryOptions.cs:51]: Added an example tag and sprinkled with `<br>s` . Let's see how it looks it docs.

**Pull request merged**